pymeasure/instruments/signalrecovery/dsp7265.py

```diff
@@ -341,29 +388,212 @@ def start_buffer(self):
+            method is called.
+        """
+        # Check if buffer is finished
+        if self.curve_buffer_status[0] != 0:
+            if wait_for_buffer:
+                self.wait_for_buffer()
             else:
-                return data
-        else:
-            return [0.0]
+                raise RuntimeError("Buffer acquisition is not yet finished.")
+
+        # Check which quantities are recorded in the buffer
+        bits = format(self.curve_buffer_bits, '021b')[::-1]
+        quantity_enums = [e for e, b in enumerate(bits) if b == "1"]
+
+        # Check if the provided quantity (if any) is indeed recorded
+        if quantity is not None:
+            if self.CURVE_BITS.index(quantity) in quantity_enums:
+                quantity_enums = [self.CURVE_BITS.index(quantity)]
+            else:
+                raise KeyError("The selected quantity '%s' is not recorded;"
+                               "quantity should be one of: %s" % (
+                                    quantity, ", ".join(
+                                        [self.CURVE_BITS[q] for q in quantity_enums]
+                                    )))
+
+        # Retrieve the data
+        data = {}
+        for enum in quantity_enums:
+            self.write("DC %d" % enum)
+            q_data = []
+
+            while True:
+                stb = format(self.adapter.connection.read_stb(), '08b')[::-1]
+
+                if bool(int(stb[2])):
+                    raise ValueError("Status byte reports command parameter error.")
+
+                if bool(int(stb[0])):
+                    break
+
+                if bool(int(stb[7])):
+                    q_data.append(int(self.read().strip()))
+
+            data[self.CURVE_BITS[enum]] = np.array(q_data)
+
+        if convert_to_float:
+            data = self.buffer_to_float(data)
+
+        if quantity is not None:
+            data = data[quantity]
+
+        return data
+
+    def buffer_to_float(self, buffer_data, sensitivity=None, sensitivity2=None, raise_error=True):
+        """
+        Method that converts the fixed-point buffer data to floating point data.
+        The provided data is converted as much as possible, but there are some
+        requirements to the data if all provided columns are to be converted; if
+        a key in the provided data cannot be converted it will be omitted in the
+        returned data or an exception will be raised, depending on the value of
+        raise_error.
+
+        The requirements for converting the data are as follows:
+        - Converting X, Y, magnitude and noise requires sensitivity data, which
+            can either be part of the provided data or can be provided via the
+            sensitivity argument
+        - The same holds for X2, Y2, magnitude2 and noise2 with sensitivity2.
+        - Converting the frequency requires both 'frequency part 1' and
+            'frequency part 2'.
```

Comment:
```suggestion
        The requirements for converting the data are as follows:
        
        - Converting X, Y, magnitude and noise requires sensitivity data, which
          can either be part of the provided data or can be provided via the
          sensitivity argument
        - The same holds for X2, Y2, magnitude2 and noise2 with sensitivity2.
        - Converting the frequency requires both 'frequency part 1' and
          'frequency part 2'.
```
I think this should fix the documentation linter messages reported at the start of the file.

Reply:
Ahh thanks, I was looking for this but didn't yet completely understand that this would be the reason for the message